### PR TITLE
[GUI] Improve window resizing behavior

### PR DESCRIPTION
**Pull request by TheButlah**
Helps mitigate https://github.com/tauri-apps/tauri/issues/4012

See https://github.com/tauri-apps/tauri/issues/4012#issuecomment-1449499149

### CODE REVIEW

File: gui/src-tauri/src/main.rs [hunk 1]
```diff
@@ -229,9 +229,15 @@ fn main() {
 			}
 			Ok(())
 		})
-		//
+		.on_window_event(|e| match e.event() {
```
Comment (ImUrX) [inline gui/src-tauri/src/main.rs:232]: Can you ``#[cfg(windows)]`` this, or test it in other OSes?

**Review by ImUrX — CHANGES_REQUESTED** [review]

File: gui/src-tauri/src/main.rs (hunk 1 above)
Comment (TheButlah) [reply gui/src-tauri/src/main.rs:232]: you can if you wish. But this is a chromium problem presumably, so I think its safer to keep it unconditional. Who knows what the OS is using as their webview.

Comment (ImUrX) [reply gui/src-tauri/src/main.rs:232]: I have had 0 issues in Linux in my experience btw

Comment (FabianLars) [reply gui/src-tauri/src/main.rs:232]: > Who knows what the OS is using as their webview.

both linux and macos currently use a webkit based webview. It shouldn't have any (positive) effect on these platforms. Buuut maybe it helps on Linux where some users see white flickering when resizing (i don't have that problem on my system so i can't test it)
tldr: `#[cfg(windows)]` should be fine, or even recommended.

**Review by ImUrX — APPROVED** [review]
